packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -453,7 +454,7 @@ export class AuthUtil {
             return state
         }
 
-        if (isBuilderIdConnection(conn) || isIdcSsoConnection(conn)) {
+        if (isBuilderIdConnection(conn) || isIdcSsoConnection(conn) || isSageMaker()) {
             if (isValidCodeWhispererCoreConnection(conn)) {
```

Comment:
I think you need to update this because the function will return incorrectly for sagemaker SSO connections. The same problem affects `isValidAmazonQConnection` function called around line 72 of this file but I wasn't able to put a comment on it.
https://github.com/aws/aws-toolkit-vscode/blob/5e8ad316718661bed35230e3596c13e5e2b53f30/packages/core/src/codewhisperer/util/authUtil.ts#L63

Reply:
I updated that function logic to ignore scopes for Sagemaker connections

Reply:
Looks to be broken still, see https://github.com/aws/aws-toolkit-vscode/pull/5858/files#r1819088749